compatibility_server/pip_checker.py

```diff
@@ -27,29 +27,44 @@
 import enum
 import json
 import logging
-import os.path
 import shlex
-import subprocess
-import tempfile
 import urllib.request
 
 from typing import Any, List, Mapping, Optional
 
+import docker
+
 PYPI_URL = 'https://pypi.org/pypi/'
+CONTAINER_WITH_PKG = "checker"
+TIME_OUT = 300  # seconds
+
+
+class PipCheckerError(Exception):
+    """Pip checker failed in an unexpected way."""
+
+    def __init__(self, error_msg: str):
+        super(PipCheckerError, self).__init__(
+            'Pip Checker failed, '
+            'logs are: {error_msg}'.format(error_msg=error_msg))
 
+        self.error_msg = error_msg
 
-class PipError(Exception):
+
+class PipError(PipCheckerError):
     """A pip command failed in an unexpected way."""
 
-    def __init__(self, command: List[str], returncode: int, stderr_path: str):
+    def __init__(self,
+                 error_msg: str,
+                 command: List[str],
+                 returncode: int):
         super(PipError, self).__init__(
-            'pip command ({command}) failed with error [{returncode}], '
-            'logs at: {stderr_path}'.format(
+            'Pip command ({command}) failed with error [{returncode}], '
+            'logs at: {error_msg}'.format(
```

Comment:
```suggestion
            '{error_msg}'.format(
```

Because error_msg isn't a file path anymore.

Reply:
Done.